query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -80,23 +80,167 @@ trait ExecPlan extends QueryCommand {
 
   protected def allTransformers: Seq[RangeVectorTransformer] = rangeVectorTransformers
 
+  // scalastyle:off method.length
+  def executeStreaming(source: ChunkSource,
```

Comment:
Ack'd you already know this, but I'll make the low-hanging-fruit note: `execute` and `executeStreaming` share lots of redundant code.

Reply:
Yes - this is very intentional. I did not want to introduce risk of regression in anyway and intentionally kept new and old code separate as much as possible. We can refactor (if really needed) in subsequent iterations. In the longer term, I do see `composeStreaming` replacing compose. We should deprecate and eventually remove the latter.